core/include/detray/propagator/navigator.hpp

```diff
@@ -141,6 +142,30 @@ class navigator {
         DETRAY_HOST_DEVICE
         scalar operator()() const { return _next->path; }
 
+        DETRAY_HOST_DEVICE
+        inline void set_unknown() { _status = navigation::status::e_unknown; }
+
+        DETRAY_HOST_DEVICE
+        inline void set_on_module() {
```

Comment:
Oh, I skipped that for now, because I don't have much time :P. So this runs after the navigator, but before some other actor (such as the stepper, I assume) that only does things when the state is ```on_module```?

Reply:
This is an actor so it runs after the stepper and navigator.
It makes sure that track propagates the given path length (one full turn in path_correction.inl) until it reaches the targeted surface. All information from navigation.update() is ignored as the candidates are cleared for every actor call. This actor is only for testing purpose.

Reply:
Hm, I think this breaks encapsulation pretty badly. I am not entirely sure what this actor is trying to solve: it seems to try to bypass the navigator altogether? Does it try to insert a fake candidate to make the stepper take a certain step without re-navigating? If possible, I would prefer writing e.g. a dummy navigator that does nothing but use an externally provided intersection instead of modifying the existing one this way